pkg/arkd-wallet-nbxplorer/core/application/wallet/service.go

```diff
@@ -0,0 +1,873 @@
+package wallet
+
+import (
+	"bytes"
+	"context"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"math"
+	"slices"
+	"strings"
+	"time"
+
+	"github.com/arkade-os/arkd/pkg/ark-lib/script"
+	"github.com/arkade-os/arkd/pkg/ark-lib/txutils"
+	"github.com/arkade-os/arkd/pkg/arkd-wallet-nbxplorer/core/application"
+	"github.com/arkade-os/arkd/pkg/arkd-wallet-nbxplorer/core/ports"
+	"github.com/btcsuite/btcd/btcec/v2"
+	"github.com/btcsuite/btcd/btcec/v2/schnorr"
+	"github.com/btcsuite/btcd/btcutil"
+	"github.com/btcsuite/btcd/btcutil/coinset"
+	"github.com/btcsuite/btcd/btcutil/psbt"
+	"github.com/btcsuite/btcd/chaincfg"
+	"github.com/btcsuite/btcd/chaincfg/chainhash"
+	"github.com/btcsuite/btcd/txscript"
+	"github.com/btcsuite/btcd/wire"
+	"github.com/btcsuite/btcwallet/waddrmgr"
+	"github.com/lightningnetwork/lnd/input"
+	"github.com/lightningnetwork/lnd/lntypes"
+	"github.com/lightningnetwork/lnd/lnwallet/chainfee"
+	log "github.com/sirupsen/logrus"
+	"github.com/tyler-smith/go-bip39"
+)
+
+var (
+	ErrWalletLocked = errors.New("wallet is locked")
+	ANCHOR_PKSCRIPT = []byte{
+		0x51, 0x02, 0x4e, 0x73,
+	}
+)
+
+// https://github.com/bitcoin/bitcoin/blob/439e58c4d8194ca37f70346727d31f52e69592ec/src/policy/policy.cpp#L23C8-L23C11
+// biggest input size to compute the maximum dust amount
+const biggestInputSize = 148 + 182 // = 330 vbytes
+
+type WalletOptions struct {
+	SeedRepository ports.SeedRepository
+	Cypher         ports.Cypher
+	Nbxplorer      ports.Nbxplorer
+	Network        string
+}
+
+type wallet struct {
+	WalletOptions
+
+	locker  *outpointLocker
+	keyMgr  *keyManager
+	isReady chan struct{}
+
+	signerPubKey *btcec.PublicKey
```

Comment:
not needed, we can compute the pubkey from the private key if we move to it from extended key in `keyMgr`